gym-unity/gym_unity/envs/__init__.py

```diff
@@ -344,6 +358,82 @@ def _check_agents(self, n_agents: int) -> None:
                 "initialization. This is not supported."
             )
 
+    def _sanitize_info(self, info: BatchedStepResult) -> BatchedStepResult:
+        self._stored_info = info  # store the new original
+        if info.n_agents() == self._n_agents:
+            return info
+        n_extra_agents = info.n_agents() - self._n_agents
+        if n_extra_agents < 0 or n_extra_agents > self._n_agents:
+            # In this case, some Agents did not request a decision when expected
+            # or too many requested a decision
+            raise UnityGymException(
+                "The number of agents in the scene does not match the expected number."
+            )
+
+        # remove the done Agents
+        indices_to_keep: List[int] = []
+        for index in range(len(info.agent_id)):
+            if not info.done[index]:
+                indices_to_keep.append(index)
+
+        # set the new AgentDone flags to True
+        for index in range(len(info.agent_id)):
+            agent_id = info.agent_id[index]
+            if not self._stored_info.contains_agent(agent_id):
```

Comment:
Is this ever true? Aren't `info` and `self._stored_info` pointing to the same thing?

Reply:
If not, I think maybe it's clearer to rewrite this loop as
```
for done_agent_id in self._done_agents:
  agent_index = info.agent_id_to_index.get(done_agent_id)
  if agent_index is not None:
    info.done[agent_index] = True
```

Reply:
Damn it

Reply:
All right updated this part.